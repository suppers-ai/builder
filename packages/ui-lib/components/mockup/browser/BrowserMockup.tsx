import { ComponentChildren } from "preact";
import { BaseComponentProps } from "../../types.ts";

export interface BrowserMockupProps extends BaseComponentProps {
  /** Child content to display in the browser */
  children: ComponentChildren;
  /** Browser title/URL */
  url?: string;
  /** Browser variant */
  variant?: "default" | "dark" | "minimal";
  /** Whether to show navigation buttons */
  showControls?: boolean;
  /** Click handler for the mockup */
  onMockupClick?: () => void;
}

export function BrowserMockup({
  children,
  url = "https://example.com",
  variant = "default",
  showControls = true,
  class: className,
  ...props
}: BrowserMockupProps) {
  // Build mockup classes
  const mockupClasses = [
    "mockup-browser",
    variant === "dark" && "bg-base-300",
    variant === "minimal" && "border-base-300",
    className,
  ].filter(Boolean).join(" ");

  return (
    <div className={mockupClasses} {...props}>
      {showControls && (
        <div className="mockup-browser-toolbar">
          <div className="input">{url}</div>
        </div>
      )}
      <div className="flex justify-center px-4 py-16 bg-base-200">
        {children}
      </div>
    </div>
  );
}
