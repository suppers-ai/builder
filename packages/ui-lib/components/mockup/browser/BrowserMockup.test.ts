import { assertEquals, assertStringIncludes } from "https://deno.land/std@0.224.0/assert/mod.ts";
import { assertSnapshot } from "https://deno.land/std@0.224.0/testing/snapshot.ts";
import { renderToString } from "preact-render-to-string";
import { BrowserMockup } from "./BrowserMockup.tsx";

Deno.test("BrowserMockup - basic rendering", () => {
  const html = renderToString(BrowserMockup({
    children: "Test content",
  }));
  assertStringIncludes(html, "mockup-browser");
  assertStringIncludes(html, "Test content");
});

Deno.test("BrowserMockup - with custom URL", () => {
  const html = renderToString(BrowserMockup({
    children: "Test content",
    url: "https://custom.example.com",
  }));
  assertStringIncludes(html, "https://custom.example.com");
});

Deno.test("BrowserMockup - without controls", () => {
  const html = renderToString(BrowserMockup({
    children: "Test content",
    showControls: false,
  }));
  assertStringIncludes(html, "mockup-browser");
  // Should not include toolbar when controls are hidden
  assertEquals(html.includes("mockup-browser-toolbar"), false);
});

Deno.test("BrowserMockup - dark variant", () => {
  const html = renderToString(BrowserMockup({
    children: "Test content",
    variant: "dark",
  }));
  assertStringIncludes(html, "bg-base-300");
});

Deno.test("BrowserMockup - minimal variant", () => {
  const html = renderToString(BrowserMockup({
    children: "Test content",
    variant: "minimal",
  }));
  assertStringIncludes(html, "border-base-300");
});

Deno.test("BrowserMockup - with custom class", () => {
  const html = renderToString(BrowserMockup({
    children: "Test content",
    class: "custom-class",
  }));
  assertStringIncludes(html, "custom-class");
});

Deno.test("BrowserMockup - HTML snapshot", async (t) => {
  const html = renderToString(BrowserMockup({
    children: "Test content for snapshot",
    url: "https://example.com",
  }));
  await assertSnapshot(t, html);
});

Deno.test("BrowserMockup - HTML snapshot dark variant", async (t) => {
  const html = renderToString(BrowserMockup({
    children: "Dark theme content",
    variant: "dark",
    url: "https://dark.example.com",
  }));
  await assertSnapshot(t, html);
});

Deno.test("BrowserMockup - HTML snapshot minimal without controls", async (t) => {
  const html = renderToString(BrowserMockup({
    children: "Minimal content",
    variant: "minimal",
    showControls: false,
  }));
  await assertSnapshot(t, html);
});
