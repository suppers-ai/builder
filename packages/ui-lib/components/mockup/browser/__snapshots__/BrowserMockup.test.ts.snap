export const snapshot = {};

snapshot[`BrowserMockupDisplay - HTML snapshot 1`] = `'<div class="mockup-browser"><div class="mockup-browser-toolbar"><div class="input">https://example.com</div></div><div class="flex justify-center px-4 py-16 bg-base-200">Test content for snapshot</div></div>'`;

snapshot[`BrowserMockupDisplay - HTML snapshot dark variant 1`] = `'<div class="mockup-browser bg-base-300"><div class="mockup-browser-toolbar"><div class="input">https://dark.example.com</div></div><div class="flex justify-center px-4 py-16 bg-base-200">Dark theme content</div></div>'`;

snapshot[`BrowserMockupDisplay - HTML snapshot minimal without controls 1`] = `'<div class="mockup-browser border-base-300"><div class="flex justify-center px-4 py-16 bg-base-200">Minimal content</div></div>'`;
